express-mutations/server.ts

```diff
@@ -0,0 +1,121 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/pagila',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.use(express.json());
+
+app.post('/api/actors', async (req, res, next) => {
+  try {
+    const { firstName, lastName } = req.body;
+    if (firstName === undefined) {
+      throw new ClientError(400, 'firstName is undefined');
+    }
+    if (lastName === undefined) {
+      throw new ClientError(400, 'lastName is undefined');
+    }
+
+    const sql = `
+    insert into "actors" ("firstName", "lastName")
+    values ($1, $2)
+    returning *
+    `;
+    const params = [firstName, lastName];
+    const results = await db.query(sql, params);
+    const post = results.rows[0];
+
+    res.status(201).json(post);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.put('/api/actors/:actorId', async (req, res, next) => {
+  try {
+    const { firstName, lastName } = req.body;
+    if (lastName === undefined) {
+      throw new ClientError(400, 'lastName is undefined');
+    } else if (firstName === undefined) {
+      throw new ClientError(400, 'firstName is undefined');
+    }
+    const { actorId } = req.params;
+    if (actorId === undefined) {
+      throw new ClientError(400, 'ActorId is undefined');
+    }
+    const sql = `
+    update "actors"
+    set
+      "firstName" = $2,
+      "lastName" = $3
+    where "actorId" = $1
+    returning *
+    `;
+    const params = [actorId, firstName, lastName];
+    const results = await db.query(sql, params);
+    const update = results.rows[0];
+    if (!update) {
+      throw new ClientError(404, `Could not load ${actorId}`);
+    }
+    res.status(200).json(update);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.delete('/api/actors/:actorId', async (req, res, next) => {
+  try {
+    const { actorId } = req.params;
+    if (actorId === undefined) {
+      throw new ClientError(400, 'ActorId is undefined');
+    }
+    const sql = `
+   delete
+    from "actors"
+    where "actorId" = $1
+    returning *
+   `;
+    const params = [actorId];
+    const results = await db.query(sql, params);
+    const deleting = results.rows[0];
+    if (!deleting) throw new ClientError(404, 'Could not load deleting');
+    res.status(204).json(deleting);
```

Comment:
With a 204 there will not be a body, so you don't need the `.json(deleting)` call. Instead, just return an empty body. There are two ways to do this:
`res.status(204).send()`
or
`res.sendStatus(204)`

The second is just a simpler way of doing the first. They both do the same thing.